src/commands/debugCommands.ts

```diff
@@ -0,0 +1,206 @@
+      PromptAnswer.YES,
+      PromptAnswer.NO,
+    );
+    if (answer === PromptAnswer.YES) {
+      try {
+        this.telemetry.sendTelemetryEvent("validateProject");
+        const project = this.dbtProjectContainer.findDBTProject(
+          projectContext.uri,
+        );
+        if (project === undefined) {
+          window.showErrorMessage(
+            `Project ${projectContext.label} was not found`,
+          );
+          return;
+        }
+        const runModelCommand = this.dbtCommandFactory.createDebugCommand(
+          project.projectRoot,
+          project.dbtProfilesDir,
+        );
+        try {
+          const runModelProcess =
+            await this.dbtProjectContainer.dbtClient?.executeCommand(
+              runModelCommand,
+            );
+          const runModelOutput = await runModelProcess.complete();
+          this.terminal.log(
+            `${runModelProcess.formatText(runModelOutput.toString())}`,
+          );
+          if (runModelOutput.includes("ERROR")) {
+            throw new Error();
+          }
+        } catch (runError) {
+          console.log(runError);
+          window.showErrorMessage(
+            "Error running dbt debug for project " +
+              projectContext.label +
+              ". Please check the output tab for more details.",
+          );
+        }
+      } catch (err) {
+        console.log(err);
+        this.telemetry.sendTelemetryError("validateProjectError", err);
+      }
+    }
+  }
+  async installDeps(projectContext: ProjectQuickPickItem | undefined) {
+    if (projectContext === undefined) {
+      // This shouldnt happen really
+      window.showErrorMessage(
+        "No project was selected, please select a project in the step 'Pick a dbt project' above.",
+      );
+      return;
+    }
+    const answer = await window.showInformationMessage(
+      `Do you want to validate the project: ${projectContext.label}? This will run the command 'dbt deps' inside this project. Do you want to continue?`,
+      PromptAnswer.YES,
+      PromptAnswer.NO,
+    );
+    if (answer === PromptAnswer.YES) {
+      try {
+        this.telemetry.sendTelemetryEvent("installDeps");
+        const project = this.dbtProjectContainer.findDBTProject(
+          projectContext.uri,
+        );
+        if (project === undefined) {
+          window.showErrorMessage(
+            `Project ${projectContext.label} was not found`,
+          );
+          return;
+        }
+
+        const depsCommand = this.dbtCommandFactory.createInstallDepsCommand(
+          project.projectRoot,
+          project.dbtProfilesDir,
+        );
+        try {
+          const dbtDepsProcess =
+            await this.dbtProjectContainer.dbtClient?.executeCommand(
+              depsCommand,
+            );
+          const depsOut = await dbtDepsProcess.complete();
```

Comment:
Why not `completeWithTerminalOutput`, that would be streaming?

Reply:
it streams to terminal and does not return the output so i cant check if it exited successfully or not. dbt debug using our command executors returns normally even if there are some tests failing or there are runtime errors.